crates/bitwarden-vault/src/vault_client.rs

```diff
@@ -5,20 +5,78 @@ use crate::{
     SyncRequest, SyncResponse,
 };
 
+/// A vault client.
+///
+/// # Examples
+///
+/// ```rust
+/// use bitwarden_core::{Client,ClientSettings,DeviceType};
+/// use bitwarden_vault::VaultClient;
+///
+/// let client = Client::new(Some(ClientSettings {
+///     identity_url: "https://identity.bitwarden.com".to_owned(),
+///     api_url: "https://api.bitwarden.com".to_owned(),
+///     user_agent: "Bitwarden Rust-SDK".to_owned(),
+///     device_type: DeviceType::ChromeBrowser,
+/// }));
+///
+/// let vault = VaultClient::new(&client);
+/// ```
```

Comment:
We don't want people to manually. wire up VaultClient's and you should generally use the Extension i.e. `client.vault() -> VaultClient`.

Reply:
@dani-garcia should we remove `pub` from the Clients?

Reply:
That sounds reasonable yeah. I looked into it and `VaultClient` is the only partial client constructor that is `pub` too, the rest are private.

Reply:
https://github.com/bitwarden/sdk-internal/pull/124